rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -410,18 +404,18 @@ void enableRecovery(@Observes @Initialized(ApplicationScoped.class) Object init)
 
     /**
      * When the deployment is unloaded unregister for recovery
-     *
-     * @param init a javax.servlet.ServletContext
      */
-    void disableRecovery(@Observes @Destroyed(ApplicationScoped.class) Object init) {
-        assert lraRecoveryModule != null;
+    @PreDestroy
+    void disableRecovery() {
+        if (lraRecoveryModule != null) {
 
-        Implementations.uninstall();
-        RecoveryManager.manager().removeModule(lraRecoveryModule, false);
-        lraRecoveryModule = null;
+            Implementations.uninstall();
```

Comment:
I note that this is currently a no-op but I wonder if a separate JBTM should be removing from the RecordTypeManager tp reverse the install?

Reply:
I provided an implementation for `Implementations.uninstall()`